### ISSUE TITLE
run_bounds_check: Windows Intel oneAPI

### ISSUE THREAD

**Issue by scivision** [gemini3d/gemini3d#81]
corresponds to #80 that covers Linux CentOS Stream 8 errors on GemCI `ctest -R run_bounds-check` on branch "ide_merge"

NOTE: Because errors like this can have true causes elsewhere in the code, perhaps fix #80 first as the runtime checks catch its location more precisely--maybe #80 is causing #81

```sh
git checkout ide_merge
```

### Windows 11 GCC 11.2

No problem, this test is OK:

```sh
ctest --test-dir build -R run_bounds_check
```

### Windows 11 Intel oneAPI 2021.4

```sh
ctest --test-dir build -R run_bounds_check
```

fails:

```
The following tests FAILED:
          6 - run_bounds_check:GDI_periodic_lowres (Failed)
         16 - run_bounds_check:arcs (Failed)
         26 - run_bounds_check:mini2dew_fang (Failed)
         36 - run_bounds_check:mini2dew_glow (Failed)
         46 - run_bounds_check:mini2dns_fang (Failed)
         56 - run_bounds_check:mini2dns_glow (Failed)
         66 - run_bounds_check:mini3d_fang (Failed)
         76 - run_bounds_check:mini3d_glow (Failed)
         86 - run_bounds_check:KHI_periodic_lowres (Failed)
         96 - run_bounds_check:cusp_softprecip2D_Dirich_CI (Failed)
        106 - run_bounds_check:cusp_softprecip2D_Neu_CI (Failed)
        116 - run_bounds_check:cusp_softprecip3D_CI (Failed)
        127 - run_bounds_check:tohoku20112D_medres_2Dneu (Failed)
        138 - run_bounds_check:tohoku20112D_medres_axineu_CI (Failed)
        149 - run_bounds_check:tohoku20113D_lowres_3Dneu (Failed)
        160 - run_bounds_check:tohoku20113D_lowres_axineu (Failed)
Errors while running CTest
```

All the errors appear similar. For comparison with #80 I include result from:

```sh
ctest --test-dir build -R run_bounds_check:tohoku20113D_lowres_axineu -FA fxt -V
```

The failure is noted in [advec_mpi.f90](https://github.com/gemini3d/gemini3d/blob/4f59e77caf4ebfe19c86feb2a37c0c64813d177f/src/numerical/advection/advec_mpi.f90#L124)
However as usual, the true error source could lie elsewhere. Usually we use `print` to further debug.

```
test 160
    Start 160: run_bounds_check:tohoku20113D_lowres_axineu

160: Test command: gemini3d\buildi\gemini3d.run.debug.exe "gemci/tohoku20113D_lowres_axineu" "-mpiexec" "C:/Program Files (x86)/Intel/oneAPI/mpi/latest/bin/mpiexec.exe" "-dryrun"
160: Environment variables:
160:  GEMINI_CIROOT=gemci
160: Test timeout computed to be: 180
160: gemini3d.run: detected CPU count: 4
160: MPI partition of lx2, lx3: 128 48 is lid2, lid3: 2 2
160: MPI images: 4
160:  "C:/Program Files (x86)/Intel/oneAPI/mpi/latest/bin/mpiexec.exe" -n 4 C:/Users/
160:  gemini3d/buildi/gemini.bin.exe gemci/tohoku20113D_low
160:  res_axineu  -dryrun
160:            4 MPI processes detected
160: gemini3d/buildi/gemini.bin.exe Process:       0 /      3 at 20211109T135415.169
160:  ******************** input config ****************
160: simulation directory: gemci/tohoku20113D_lowres_axineu
160:                             start year-month-day:    2011-03-11
160:                                       start time:   35100.000
160:                                         duration:    1795.000
160:                                     output every:     125.000
160: gemini.f90: using input data files:
160: gemci/tohoku20113D_lowres_axineu/inputs/simsize.h5
160: gemci/tohoku20113D_lowres_axineu/inputs/simgrid.h5
160: gemci/tohoku20113D_lowres_axineu/inputs/initial_conditions.h5
160:  Neutral disturbance mlat,mlon:     29.0667000000000
160:    209.850100000000
160:  Neutral disturbance cadence (s):     5.00000000000000
160:  Neutral grid resolution (m):     8333.30000000000        5000.00000000000
160:  Neutral disturbance data files located in directory:
160:  gemci/acoustic2D_axisymmetric_neutrals
160:  no precipitation specified
160:  no Efield specified
160:  GLOW disabled
160:  MSISE00 enabled for neutral atmosphere calculations.
160:  EIA disabled
160:  Variable background neutral atmosphere disabled.
160:  Background precipitation has total energy flux and energy:
160:   1.000000000000000E-003   3000.00000000000
160:  Parallel current calculation enabled.
160:  Inertial capacitance calculation type:             0
160:  Diffusion solve type:             2
160:  Milestone output selected; cadence (every nth output) of:            10
160:  Gravitaional drift terms disabled.
160:  Lagrangian grid disabled
160:  **************** end input config ***************
160:  grid_size_root: full grid size:           512         128          48
160: process grid (Number MPI processes) x2, x3:  2 2
160: Process:0 at process grid location: 0 0
160:  get_subgrid_size: 3D run
160:   Detected dipole grid...
160: make_dipolemesh:  allocating space for grid of size: 516 68 28
160:  make_dipolemesh:  converting cell centers to spherical coordinates...
160: gemini3d/buildi/gemini.bin.exe Process:       1 /      3 at 20211109T135415.185
160: gemini3d/buildi/gemini.bin.exe Process:       2 /      3 at 20211109T135415.185
160: gemini3d/buildi/gemini.bin.exe Process:       3 /      3 at 20211109T135415.185
160:  make_dipolemesh:  converting cell interfaces in q...
160:  make_dipolemesh:  converting cell interfaces in p...
160: forrtl: severe (157): Program Exception - access violation
160: Image              PC                Routine            Line        Source
160: impi.dll           00007FFB706CFFF4  Unknown               Unknown  Unknown
160: impi.dll           00007FFB70D0623F  Unknown               Unknown  Unknown
160: gemini.bin.exe     00007FF7C9EB9780  Unknown               Unknown  Unknown
160: gemini.bin.exe     00007FF7C9EB9155  Unknown               Unknown  Unknown
160: gemini.bin.exe     00007FF7CA184601  Unknown               Unknown  Unknown
160: gemini.bin.exe     00007FF7C98B6283  ADVEC_MPI_mp_ADVE         124  advec_mpi.f90
160: gemini.bin.exe     00007FF7CB5E1F7E  Unknown               Unknown  Unknown
160: gemini.bin.exe     00007FF7CC298559  Unknown               Unknown  Unknown
160: gemini.bin.exe     00007FF7CC2983FE  Unknown               Unknown  Unknown
160: gemini.bin.exe     00007FF7CC2982BE  Unknown               Unknown  Unknown
160: gemini.bin.exe     00007FF7CC2985EE  Unknown               Unknown  Unknown
```

**Comment by mattzett** [comment]
I'm getting a similar issue with intel on ubuntu 20.04 except that it fails the mumps test oddly...

```
-- The C compiler identification is IntelLLVM 2021.4.0
-- The Fortran compiler identification is Intel 2021.4.0.20210910
```

When I compile with optimization it is fine; when I use the debug preset I get stuff like this:

```
28:  Effective threshold for pivoting, CNTL(1)  =      0.1000D-01
28: forrtl: error (65): floating invalid
28: Image              PC                Routine            Line        Source             
28: test_mumps         0000000000849C4B  Unknown               Unknown  Unknown
28: libpthread-2.31.s  00007FCE4C59E3C0  Unknown               Unknown  Unknown
28: test_mumps         000000000065D279  Unknown               Unknown  Unknown
28: test_mumps         0000000000658896  Unknown               Unknown  Unknown
28: test_mumps         0000000000656060  Unknown               Unknown  Unknown
28: test_mumps         00000000005D451B  Unknown               Unknown  Unknown
28: test_mumps         000000000040AC61  Unknown               Unknown  Unknown
28: test_mumps         000000000040772D  test_mumps_IP_sim         106  test_mumps.f90
28: test_mumps         0000000000405FD2  MAIN__                     25  test_mumps.f90
28: test_mumps         0000000000405F22  Unknown               Unknown  Unknown
28: libc-2.31.so       00007FCE4A3B80B3  __libc_start_main     Unknown  Unknown
28: test_mumps         0000000000405E2E  Unknown               Unknown  Unknown
28: 
28: ===================================================================================
28: =   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
28: =   RANK 0 PID 89733 RUNNING AT parallels-Parallels-Virtual-Platform
28: =   KILLED BY SIGNAL: 6 (Aborted)
28: ===================================================================================
28: 
28: ===================================================================================
28: =   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
28: =   RANK 1 PID 89734 RUNNING AT parallels-Parallels-Virtual-Platform
28: =   KILLED BY SIGNAL: 9 (Killed)
28: ===================================================================================
3/3 Test #28: unit:mumps_basic .................***Failed    0.88 sec
```

**Comment by mattzett** [comment]
both icc and icx cause this same issue

**Comment by scivision** [comment]
Matt's error here is unrelated, it was due to `-fpe0` that we have subsequently disabled.

This error I noted is peculiar to Windows and we don't have a actionable answer yet as to why it's happening. 
We are deferring any action for now.

To track it down, using `git bisect` may be handy.

**Comment by mattzett** [comment]
Yep thanks for clarifying

**Comment by scivision** [comment]
This could be a false positive. Since it's also on "main" branch, it would require twiddling with compiler flags one at a time.
Starting with same flags as Release (no manually added debug checks) and then add flags one at a time.

However, since the only flag that's added for Intel complier is `-CB` (recognized on Windows as `/CB`) this might also be futile. I removed the `-debug` flag and it still crashed the same way.

So I'm writing it off as a Windows Intel bug in debug mode only.

**Comment by scivision** [comment]
this issue was fixed by 5cd1b3deb51645d49b0cf36c857df11732eb7bb0
